parlai/agents/fairseq/fairseq.py

```diff
@@ -401,45 +401,29 @@ def reset(self):
         super().reset()
         self.reset_metrics()
 
-    def batch_act(self, observations):
-        bsz = len(observations)
-        # initialize a table of replies with this agent's id
-        batch_reply = [{"id": self.getID()} for _ in range(bsz)]
-
-        # torchagent boilerplate
-        self.is_training = any(["labels" in obs for obs in observations])
-        vec_obs = [self.vectorize(obs) for obs in observations]
-        xs, _, ys, _, valid_inds = self.map_valid(vec_obs)
-        if xs is None:
-            return batch_reply
-
-        # here begins fairseq specific stuff
-        samples = self._make_sample(xs, ys)
+    def batchify(self, *args, **kwargs):
```

Comment:
So I don't know of a reason that `fairseq` needs `sort=True`. That was the default behavior from @vgonzalez88 because something else expected the batch sorted, but I can't remember the details.

Reply:
Ahh, yes, sadly the fairseq lstm is using this (but not Transformer). Thanks for checking into this.